SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -280,6 +281,69 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	}
 }
 
+void USpatialNetDriver::CreateServerWorkerEntity(int AttemptCounter)
+{
+	const WorkerRequirementSet WorkerIdPermission{ { FString::Format(TEXT("workerId:{0}"), { Connection->GetWorkerId() }) } };
+
+	WriteAclMap ComponentWriteAcl;
+	ComponentWriteAcl.Add(SpatialConstants::POSITION_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::METADATA_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, WorkerIdPermission);
+	ComponentWriteAcl.Add(SpatialConstants::INTEREST_COMPONENT_ID, WorkerIdPermission);
+
+	QueryConstraint constraint;
+	constraint.EntityIdConstraint = SpatialConstants::INITIAL_GLOBAL_STATE_MANAGER_ENTITY_ID;
+
+	Query query;
+	query.Constraint = constraint;
+	query.FullSnapshotResult = true;
+
+	ComponentInterest queries;
+	queries.Queries.Add(query);
+
+	Interest interest;
+	interest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, queries);
+
+	TArray<Worker_ComponentData> Components;
+	Components.Add(Position().CreatePositionData());
+	Components.Add(Metadata(FString::Format(TEXT("WorkerEntity:{0}"), { Connection->GetWorkerId() })).CreateMetadataData());
+	Components.Add(EntityAcl(WorkerIdPermission, ComponentWriteAcl).CreateEntityAclData());
+	Components.Add(interest.CreateInterestData());
+
+	Worker_RequestId RequestId = Connection->SendCreateEntityRequest(MoveTemp(Components), nullptr);
+
+	CreateEntityDelegate OnCreateWorkerEntityResponse;
+	OnCreateWorkerEntityResponse.BindLambda([this, AttemptCounter](const Worker_CreateEntityResponseOp& Op)
+	{
+		if (Op.status_code == WORKER_STATUS_CODE_SUCCESS)
+		{
+			return;
+		}
+
+		if (Op.status_code != WORKER_STATUS_CODE_TIMEOUT)
+		{
+			UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Worker entity creation request failed: \"%s\""),
+				UTF8_TO_TCHAR(Op.message));
+			return;
+		}
+
+		if (AttemptCounter == SpatialConstants::MAX_NUMBER_COMMAND_ATTEMPTS)
+		{
+			UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Worker entity creation request timed out too many times. (%u attempts)"),
+				SpatialConstants::MAX_NUMBER_COMMAND_ATTEMPTS);
+			return;
+		}
+
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Worker entity creation request timed out and will retry."));
+		FTimerHandle RetryTimer;
+		TimerManager.SetTimer(RetryTimer, [this, AttemptCounter]()
+		{
+			CreateServerWorkerEntity(AttemptCounter + 1);
+		}, SpatialConstants::GetCommandRetryWaitTimeSeconds(AttemptCounter), false);
+	});
+	Dispatcher->AddCreateEntityDelegate(RequestId, OnCreateWorkerEntityResponse);
```

Comment:
Hmmmmmmmmmmmmm, I'm really against this. The dispatcher is one of those classes that we've sort of encapsulated quite well and only use it for one purpose. I think you should register this delegate with the Receiver.